### PR TITLE
Allow explicit SDRF provision

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR makes the SDRF condense processes parameterised for incorporation into other workflows. Specifically, IDF/SDRF locations can be inferred as previously (from ATLAS_PROD / ATLAS_SC_EXPERIMENTS) or by specific arguments to the two changed scripts (condense_sdrf.pl, single_cell_condensed_sdrf.sh).

I've also tidied up for consistency. Previously, single_cell_condensed_sdrf.sh would infer SDRF locations for its own purposes, and condense_sdrf.pl would do the same using its own rules. Now, when single_cell_condensed_sdrf.sh derives the annotation paths, these are passed explicitly to condense_sdrf.pl. Similarly for the zooma exclusions, previously derived independently by the two scripts. 

Changes /should/ be backwards compatible.

I've also added a test suite.

### CODE REVIEW

File: single_cell_condensed_sdrf.sh [hunk 1]
```diff
@@ -130,31 +193,39 @@ if [ "$replicates" = true ]; then
...
+export CONDENSED_SDRF_TSV=$outputDir/$expId.condensed-sdrf.tsv
 # Explode condensed SDRF for droplet experiments from RUN_ID to RUN_ID-CELL_ID.
-if [ -f $ATLAS_SC_EXPERIMENTS/$EXP_ID/cell_to_library.txt ]; then
+if [ -f $cellToLib ]; then
   use_run_id_cell_id_In_condensed
 fi
 
 # Explode condensed SDRF with inferred cell type
-export CELLTYPES=$ATLAS_PROD/singlecell/experiment/$EXP_ID.cells.txt
+export CELLTYPES=$ATLAS_PROD/singlecell/experiment/$expId.cells.txt
```
Comment (pcm32) [inline single_cell_condensed_sdrf.sh]: Probably this needs to follow the same spirit of resolving the file based on where the IDF is.

File: single_cell_condensed_sdrf.sh [hunk 2]
```diff
@@ -130,31 +193,39 @@ if [ "$replicates" = true ]; then
...
-  echo "Found cell types file for $EXP_ID"
+  echo "Found cell types file for $expId"
   use_cell_types_In_condensed
+
+
   annotate_celltypes_condensed_sdrf.pl -c $CONDENSED_SDRF_TSV \
                                        -o $CONDENSED_SDRF_TSV"_celltypes" \
-                                       -l $CONDENSED_SDRF_TSV"_zoomalogs" \
-                                       -x $scriptDir/../supporting_files/zooma_exclusions.yml
+                                       -l $CONDENSED_SDRF_TSV"_zoomalogs"$exclusions
```
Comment (pcm32) [inline single_cell_condensed_sdrf.sh]: For clarity I would go for:
```suggestion
                                       -l $CONDENSED_SDRF_TSV"_zoomalogs" \
                                       $exclusions
```
although I guess that a line break with an empty string could be problematic?

File: single_cell_condensed_sdrf.sh (hunk 1 above)
Comment (pinin4fjords) [reply single_cell_condensed_sdrf.sh]: Done!

File: single_cell_condensed_sdrf.sh (hunk 2 above)
Comment (pinin4fjords) [reply single_cell_condensed_sdrf.sh]: Sorry, did it manually without thinking (rather than accepting your suggestion). But resolved.

**Comment by pinin4fjords** [comment]
Thanks Pablo. As I say, I've done my best to make sure the new code works both ways (see the 'implict' tests). but as you say, care required.

**Pull request merged**